src/asciidoctor-core-api.js

```diff
@@ -1465,7 +1465,61 @@ Cursor.prototype.getLineNumber = function () {
 };
 
 // Logger API (available in Asciidoctor 1.5.7+)
-// REMIND: we are using "skip_missing" because this API is only available starting with Asciidoctor 1.5.7
+
+function initializeLoggerFormatterClass (className, functions) {
+  var superclass = Opal.const_get_qualified(Opal.Logger, 'Formatter');
+  var scope = Opal.klass(Opal.Object, superclass, className, function () {});
+  var postConstructFunction;
+  var initializeFunction;
+  for (var key in functions) {
+    if (functions.hasOwnProperty(key)) {
+      (function (key) {
+        var userFunction = functions[key];
+        if (key === 'postConstruct') {
+          postConstructFunction = userFunction;
+        } else if (key === 'initialize') {
+          initializeFunction = userFunction;
+        } else {
+          Opal.def(scope, '$' + key, function () {
+            for (var i = 0; i < arguments.length; i++) {
```

Comment:
Here we tranform Opal.Hash to JSON for usability.
Though it's actually only really useful for the `message` argument in the `call` function.